adhocteam/Head-Start-TTADP
# Create separate docker-compose file for running tests, add bin/run-tests command for easy testing

**Description of change**

While looking at issues with the backend tests, I saw some room for improvement with how docker tests run locally. The `bin/run-tests` file is the culmination of my attempts to sandbox docker-tests from docker-dev environment. 

I tried [multiple docker-compose.yml](https://docs.docker.com/compose/extends/) files, but that had many issues. For one, overrides don't override multivalve options. Another issues was Docker not treating the different configurations as different services, so it wasn't using the test-db when expected. 

I looked at the `circleci` CLI, but that only runs jobs, not workflows, and the circleci needed to run tests is two jobs. There may be something useful here, but also introduces another tool.

If you look in the commit history, you can also see that I wrote a much longer bash script initially, in part to help me experiment with multiple docker-compose.yml files and in part to compose a number of smallish docker commands into larger actions. When I realized multiple docker-compose.yml files wasn't going to work, this tool became unnecessary. You could retrieve it via `git checkout 8b0beced865b3a9733bb5c61123dcc9d33451981^ -- bin/docker-run` if you're interested.

**How to test**

Run `./bin/run-tests` then start your regular Docker environment. Running tests should not overwrite/modify your development database.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/263


**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
